src/agent/coverage/src/sancov.rs

```diff
@@ -0,0 +1,429 @@
+                self.llvm_pcs_start = offset;
+            }
+            LlvmPcsStop => {
+                self.llvm_pcs_stop = offset;
+            }
+            MsvcBoolsStart => {
+                self.msvc_bools_start = offset;
+            }
+            MsvcBoolsStop => {
+                self.msvc_bools_stop = offset;
+            }
+            MsvcCountersStart => {
+                self.msvc_counters_start = offset;
+            }
+            MsvcCountersStop => {
+                self.msvc_counters_stop = offset;
+            }
+            MsvcPcsStart => {
+                self.msvc_pcs_start = offset;
+            }
+            MsvcPcsStop => {
+                self.msvc_pcs_stop = offset;
+            }
+            MsvcPreviewCountersStart => {
+                self.msvc_preview_counters_start = offset;
+            }
+            MsvcPreviewCountersStop => {
+                self.msvc_preview_counters_stop = offset;
+            }
+        }
+    }
+}
+
+/// A table of Sancov instrumentation data.
+///
+/// It is an array of either bytes or (packed pairs of) pointer-sized integers.
+#[derive(Clone, Copy, Debug, Eq, PartialEq)]
+pub struct SancovTable {
+    pub ty: SancovTableTy,
+
+    /// Module-relative offset of the first array element.
+    pub offset: u32,
+
+    /// Size of the array region (in bytes).
+    ///
+    /// For `u8`-sized elements, this is also the length, but for PC tables,
+    /// this will be the product of the length and entry count, where each
+    /// entry is defined in LLVM as:
+    ///
+    /// ```c
+    /// struct PCTableEntry {
+    ///   uintptr_t PC, PCFlags;
+    /// };
+    /// ```
+    pub size: usize,
+}
+
+impl SancovTable {
+    pub fn range(&self) -> std::ops::Range<u32> {
+        self.offset..(self.offset + (self.size as u32))
+    }
+}
+
+#[derive(Clone, Copy, Debug, Eq, PartialEq)]
+pub enum SancovTableTy {
+    Bools,
+    Counters,
+    Pcs,
+}
+
+/// Note: on Windows, the LLVM `__start_` delimiter symbols do not denote the
+/// first entry of a Sancov table array, but an anchor offset that precedes it
+/// by 8 bytes.
+///
+/// See:
+/// - `compiler-rt/lib/sanitizer_common/sanitizer_coverage_win_sections.cpp`
+/// - `ModuleSanitizerCoverage::CreateSecStartEnd()` in
+///   `llvm/lib/Transforms/Instrumentation/SanitizerCoverage.cpp:350-351`
+#[derive(Clone, Copy, Debug, PartialEq, Eq)]
+pub enum Delimiter {
+    LlvmBoolsStart,
+    LlvmBoolsStop,
+    LlvmCountersStart,
+    LlvmCountersStop,
+    LlvmPcsStart,
+    LlvmPcsStop,
+    MsvcBoolsStart,
+    MsvcBoolsStop,
+    MsvcCountersStart,
+    MsvcCountersStop,
+    MsvcPcsStart,
+    MsvcPcsStop,
+    MsvcPreviewCountersStart,
+    MsvcPreviewCountersStop,
+}
+
+/// Size of padding inserted (on Window) between `__start_` delimiter symbols
+/// and the first entry of the delimited table's array.
+///
+/// To find the true start offset of the table, add this to the symbol value.
+const DELIMITER_START_PADDING: u32 = 8;
```

Comment:
Can we move this earlier in the file?  I thought you missed it in the reorg when reviewing define_table_getter